### PR TITLE
new resource: `azurerm_private_dns_resolver_forwarding_rule`

### PR DESCRIPTION
**Pull request by ms-henglu**
https://github.com/hashicorp/terraform-provider-azurerm/issues/16956

image

### CODE REVIEW

**Review by katbyte — CHANGES_REQUESTED** [review]
have an odd test failure:
```
------- Stdout: -------
=== RUN   TestAccPrivateDNSResolverForwardingRule_requiresImport
=== PAUSE TestAccPrivateDNSResolverForwardingRule_requiresImport
=== CONT  TestAccPrivateDNSResolverForwardingRule_requiresImport
    testcase.go:110: Step 1/2 error: Error running apply: exit status 1
        
        Error: creating Outbound Endpoint (Subscription: "*******"
        Resource Group Name: "acctest-rg-221031233116378887"
        Dns Resolver Name: "acctest-dr-221031233116378887"
        Outbound Endpoint Name: "acctest-droe-221031233116378887"): polling after CreateOrUpdate: Code="InternalServerError" Message="Operation has exceeded maximum processing count. outboundEndpointResourceId=/subscriptions/*******/resourceGroups/acctest-rg-221031233116378887/dnsResolvers/acctest-dr-221031233116378887/outboundEndpoints/acctest-droe-221031233116378887, circuitBreaker=[state=Open, processedCount=6, maxProcessedCount=5]"
        
          with azurerm_private_dns_resolver_outbound_endpoint.test,
          on terraform_plugin_test.tf line 41, in resource "azurerm_private_dns_resolver_outbound_endpoint" "test":
          41: resource "azurerm_private_dns_resolver_outbound_endpoint" "test" {
        
        creating Outbound Endpoint (Subscription:
        "*******"
        Resource Group Name: "acctest-rg-221031233116378887"
        Dns Resolver Name: "acctest-dr-221031233116378887"
        Outbound Endpoint Name: "acctest-droe-221031233116378887"): polling after
        CreateOrUpdate: Code="InternalServerError" Message="Operation has exceeded
        maximum processing count.
        outboundEndpointResourceId=/subscriptions/*******/resourceGroups/acctest-rg-221031233116378887/dnsResolvers/acctest-dr-221031233116378887/outboundEndpoints/acctest-droe-221031233116378887,
        circuitBreaker=[state=Open, processedCount=6, maxProcessedCount=5]"
    testing_new.go:84: Error running post-test destroy, there may be dangling resources: exit status 1
        
        Error: deleting Subnet: (Name "outbounddns" / Virtual Network Name "acctest-rg-221031233116378887" / Resource Group "acctest-rg-221031233116378887"): network.SubnetsClient#Delete: Failure sending request: StatusCode=400 -- Original Error: Code="InUseSubnetCannotBeDeleted" Message="Subnet outbounddns is in use by /subscriptions/*******/resourceGroups/acctest-rg-221031233116378887/providers/Microsoft.Network/virtualNetworks/acctest-rg-221031233116378887/subnets/outbounddns/serviceAssociationLinks/dnsResolverLink and cannot be deleted. In order to delete the subnet, delete all the resources within the subnet. See aka.ms/deletesubnet." Details=[]
        
        
        Error: deleting Dns Resolver (Subscription: "*******"
        Resource Group Name: "acctest-rg-221031233116378887"
        Dns Resolver Name: "acctest-dr-221031233116378887"): performing Delete: dnsresolvers.DnsResolversClient#Delete: Failure sending request: StatusCode=0 -- Original Error: autorest/azure: Service returned an error. Status=<nil> Code="CannotDeleteResource" Message="Can not delete resource before nested resources are deleted."
        
        deleting Dns Resolver (Subscription: "*******"
        Resource Group Name: "acctest-rg-221031233116378887"
        Dns Resolver Name: "acctest-dr-221031233116378887"): performing Delete:
        dnsresolvers.DnsResolversClient#Delete: Failure sending request: StatusCode=0
        -- Original Error: autorest/azure: Service returned an error. Status=<nil>
        Code="CannotDeleteResource" Message="Can not delete resource before nested
        resources are deleted."
--- FAIL: TestAccPrivateDNSResolverForwardingRule_requiresImport (734.67s)
```

**Comment by ms-henglu** [comment]
Hi @katbyte , 

Tests passed!
image

File: internal/services/privatednsresolver/private_dns_resolver_forwarding_rule_resource.go [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+func (r PrivateDNSResolverForwardingRuleResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"private_dns_resolver_dns_forwarding_ruleset_id": {
```
Comment (katbyte) [inline internal/services/privatednsresolver/private_dns_resolver_forwarding_rule_resource.go:55]: seems like this should be
```suggestion
		"dns_forwarding_ruleset_id": {
```

**Review by katbyte — CHANGES_REQUESTED** [review]
THanks @ms-henglu - one comment left and then i think this should be good to go

**Comment by ms-henglu** [comment]
Hi @katbyte - I also made this change to `azurerm_private_dns_resolver_virtual_network_link` resource, please check. Thanks!

**Review by katbyte — APPROVED** [review]
thanks @ms-henglu - LGTM ☁️

**Pull request merged**